web3/manager.py

```diff
@@ -75,6 +75,16 @@
 
 
 NULL_RESPONSES = [None, HexBytes("0x"), "0x"]
+METHOD_NOT_FOUND = -32601
+
+
+def _raise_bad_response_format(response: RPCResponse, error: str = "") -> None:
+    error_message = f"The error is: {error}. " if error else ""
+    raise BadResponseFormat(
```

Comment:
+1 for readability
In terms of including additional arguments for the response, it would be great to include as much context as possible. Opened a new issue (#3094) to consider revamping for v7.

Reply:
Love that. If it's just new features, not sure how much of that would need to be exclusive to `v7` too. Just adding the option to include more information on our exceptions would be great though. I think we've talked about this in the past but this issue should help track that 👌🏼